src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1323,4 +1323,13 @@
   <data name="PreferConstCharOverConstUnitStringInStringBuilderTitle" xml:space="preserve">
     <value>Consider using 'StringBuilder.Append(char)' when applicable.</value>
   </data>
-</root>
+  <data name="PreferIsEmptyOverCountDescription" xml:space="preserve">
+    <value>Prefer IsEmpty over Count when checking if your instance is empty.</value>
```

Comment:
```suggestion
    <value>Prefer 'IsEmpty' over 'Count' when checking if your instance is empty.</value>
```

Reply:
I think we need to add more to the description text explaining the underlying reason behind the suggestion, i.e. what is the performance benefit from making the suggested change.

Reply:
Here's a suggestion taken from the [`ConcurrentStack<T>`]() documentation:

_"For determining whether the type contains any items, prefer using 'IsEmpty' property rather than retrieving the number of items from the 'Count' property and comparing it to 0."_

cc @terrajobst

Reply:
I was expecting something such as “int comparison is more expensive then bool comparison”, in better wording though.

Reply:
As mentioned in https://github.com/dotnet/runtime/issues/33801#issuecomment-617841271.

In some cases, like in concurrent collections, it is faster to use `IsEmpty` since the getter will stop at the first item, similar to the `Count()` vs `Any()` rule. 
In many other cases it just reads better. 

With that said, I don't think that we should say that comparing `bool` vs `int` is always faster, I don't know if it is significant enough to be mentioned, 

There's also cases where IsEmpty looks into Count:
```
public sealed partial class ImmutableHashSet<T>{
    public bool IsEmpty
    {
        get { return this.Count == 0; }
    }
}
```

I think that the purpose of the rule was to use `IsEmpty` when there was a clear intent of "check if the object contains or does not contain items".

Was your suggestion aiming to be brief? If that's the case, I don't think it applies here, since there is similar wording on `DoNotUseCountWhenAnyCanBeUsedDescription`
https://github.com/dotnet/roslyn-analyzers/blob/4d69cb3c3abd19f1d5f1308bcd4febf1432400d6/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx#L222-L224

Reply:
Thanks for the clarification! I think what you have is fine.